src/MiniJuvix/Syntax/MicroJuvix/Language/Extra.hs

```diff
@@ -0,0 +1,323 @@
+module MiniJuvix.Syntax.MicroJuvix.Language.Extra
+  ( module MiniJuvix.Syntax.MicroJuvix.Language.Extra,
+    module MiniJuvix.Syntax.MicroJuvix.Language,
+  )
+where
+
+import Data.HashMap.Strict qualified as HashMap
+import MiniJuvix.Prelude
+import MiniJuvix.Syntax.MicroJuvix.Language
+
+data Caller
+  = CallerInductive InductiveName
+  | CallerFunction FunctionName
+  | CallerAxiom AxiomName
+  deriving stock (Eq, Ord, Generic)
+
+data TypeCallIden
+  = InductiveIden InductiveName
+  | FunctionIden FunctionName
+  deriving stock (Eq, Ord, Generic)
+
+data TypeCall' a = TypeCall'
+  { _typeCallIden :: TypeCallIden,
+    _typeCallArguments :: NonEmpty a
+  }
+  deriving stock (Eq, Generic)
+
+newtype TypeCallsMap = TypeCallsMap
+  { _typeCallsMap :: HashMap Caller (HashSet TypeCall)
+  }
+
+instance Functor TypeCall' where
+  fmap f (TypeCall' i args) = TypeCall' i (fmap f args)
+
+newtype ConcreteType = ConcreteType {_unconcreteType :: Type}
+  deriving stock (Eq, Generic)
+
+type ConcreteTypeCall = TypeCall' ConcreteType
+
+type TypeCall = TypeCall' Type
+
+type SubsE = HashMap VarName Expression
+
+type Rename = HashMap VarName VarName
+
+type Subs = HashMap VarName Type
```

Comment:
Here and below, what does the word `Subs` mean in the names?

Reply:
Substitution